client/me/privacy/main.jsx

```diff
@@ -135,9 +173,45 @@ const Privacy = createReactClass( {
 						</FormButton>
 					</form>
 				</Card>
+				<SectionHeader label={ translate( 'Legal' ) } />
+				<Card>
+					<p>
+						{ translate(
+							'A Data Processing Addendum (DPA) allows web sites and companies to assure customers, vendors, ' +
+								'and partners that their data handling complies with the law.'
+						) }
+					</p>
+
+					<p>{ translate( 'Note: most free site owners or hobbyists do not need a DPA.' ) }</p>
+
+					<p>
+						<strong>
+							{ translate(
+								'Having a DPA does not change any of our privacy and security practices for site visitors. ' +
+									'Everyone using our service gets the same high standards of privacy and security.'
+							) }
+						</strong>
+					</p>
+					<Button primary className="privacy__dpa-request-button" onClick={ this.requestDpa }>
+						{ translate( 'Request a DPA' ) }
+					</Button>
+				</Card>
 			</Main>
 		);
 	},
 } );
 
-export default compose( localize, protectForm )( Privacy );
+export default compose(
+	localize,
+	protectForm,
+	connect(
+		() => {
+			const dpaRequest = getHttpData( `dpa-request` );
+			return {
+				requestingDpa: dpaRequest.state === 'pending',
+				dpaRequestError: dpaRequest.state === 'failure' ? dpaRequest.error : undefined,
+			};
```

Comment:
Based on the usage in the component it may not care about this much detail in the request. You can consider passing in a custom state descriptor here that will form the more solid API for the UI component

```js
const dpaRequestState = request => {
  switch (request.status) {
    case 'success':
      return [ 'success' ];
    case 'failure':
      return [ 'failure', request.error ];
    default:
      return [ 'unsent' ];
  }
}
```

With this the component can focus on just the aspects of that request it cares about and the adapter can separate the logic - I find this helpful

Reply:
I initially liked the array approach, but based on usage and the fact that I'm less comfortable with index-based things without type-checking, I took this idea but returned objects instead because I judged named values to be clearer and more explicit.